src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -186,25 +222,22 @@ public boolean publish(String message, JSONArray attachments, String color) {
     }
 
     private String getTokenToUse() {
-        if (authTokenCredentialId != null && !authTokenCredentialId.isEmpty()) {
-            StringCredentials credentials = lookupCredentials(authTokenCredentialId);
-            if (credentials != null) {
-                logger.fine("Using Integration Token Credential ID.");
-                return credentials.getSecret().getPlainText();
+        if (populatedToken != null) {
+            return populatedToken;
+        } else {
+            if (!StringUtils.isEmpty(authTokenCredentialId)) {
+                StringCredentials credentials = CredentialsObtainer.lookupCredentials(authTokenCredentialId, null);
+                if (credentials != null) {
+                    logger.fine("Using Integration Token Credential ID.");
+                    return credentials.getSecret().getPlainText();
+                }
             }
-        }
-
-        logger.fine("Using Integration Token.");
 
+            logger.fine("Using Integration Token.");
```

Comment:
this doesn't throw an exception if its not found, 
Could this be moved to the legacy constructors instead?
and add the throwing of the exception?